internal/work/dispatcher.go

```diff
@@ -328,3 +328,26 @@ func (d *Dispatcher) senderName(sender dbus.Sender) (string, error) {
 
 	return "", fmt.Errorf("cannot get name for sender: %v", sender)
 }
+
+// CancelMessage implements the dispatch of a cancel method to the worker.
+func (d *Dispatcher) CancelMessage(command yggdrasil.Command, id string) error {
+	// Send the message through the cancel interface
+	directive := command.Arguments["directive"]
```

Comment:
Shouldn't we check the existence of `directive` as we do it for `messageID`?

Reply:
You are right :smile: